### DISCUSSION TITLE
Previous orders for guest email that transitions to create an account

### DISCUSSION THREAD

**Discussion by carlosa54** [saleor/saleor#8161] [Q&A]
Hi,

I'm testing saleor and I've created a couple of orders with a guest email that I later created an account for using the `accountRegister` mutation. When I try to see/fetch the previous orders I'm only getting the ones that I created while being logged in and not the previous ones.

I searched for any "previous order" logic in the saleor repo but didn't find any. If someone can point me in the right direction.

![image](https://user-images.githubusercontent.com/4250593/134264978-bdf7d6e3-37d5-409d-ad10-21e4d3fe1162.png)

Referencing the docs [here](https://docs.saleor.io/docs/3.0/developer/users#previous-orders)

**Comment by carlosa54** [comment]
Saw this block of code in the `OrderUpdate` mutation that will attach the user but requires an explicit update to an order. Not working as described in the docs
https://github.com/mirumee/saleor/blob/a63467b7a4fecf1b3c4a6307529e151aa1925d72/saleor/graphql/order/mutations/orders.py#L220-L226

**Reply by karolkielecki** [reply]
Thanks for reporting. We'll investigate it as soon as we can. Cheers!
